src/test/cls_refcount/test_cls_refcount.cc

```diff
@@ -133,7 +133,6 @@ TEST(cls_rgw, test_put_snap) {
   ASSERT_EQ(0, ioctx.snap_create("snapbar"));
 
   librados::ObjectWriteOperation *op = new_op();
-  op->create(false);
```

Comment:
if you getxattr on a whiteout, ObjectStore returns ENODATA instead of ENOENT.  and the cls code isn't differentiating between the two (for annoying convoluted reasons). 

but yeah, there is a general problem with whiteouts that cls getxattr returns ENODATA instead of ENOENT.  I started a patch that wrapped the do_osd_ops read calls in objclass.cc with a check for exists || is_whiteout() (the osd does this is do_op or something else above do_osd_op; strangely a few osd ops do it explicitly too) but i guess I didn't include it here.  :(

either way, it's not a new problem.. it's a general whiteout issue.  I'll open a separate bug for it and run this through rgw/rbd in the meantime?